features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootPresenter.kt

```diff
@@ -17,23 +17,71 @@
 package io.element.android.features.createroom.impl.root
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
 import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.matrix.api.core.MatrixPatterns
+import io.element.android.libraries.matrix.api.core.UserId
+import io.element.android.libraries.matrix.ui.model.MatrixUser
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toImmutableList
+import timber.log.Timber
 import javax.inject.Inject
 
 class CreateRoomRootPresenter @Inject constructor() : Presenter<CreateRoomRootState> {
 
     @Composable
     override fun present(): CreateRoomRootState {
+        var searchQuery by rememberSaveable { mutableStateOf("") }
+        val searchResults: MutableState<ImmutableList<MatrixUser>> = remember {
+            mutableStateOf(persistentListOf())
+        }
 
         fun handleEvents(event: CreateRoomRootEvents) {
             when (event) {
+                is CreateRoomRootEvents.UpdateSearchQuery -> searchQuery = event.query
+                is CreateRoomRootEvents.StartDM -> handleStartDM(event.matrixUser)
                 CreateRoomRootEvents.CreateRoom -> Unit // Todo Handle create room action
                 CreateRoomRootEvents.InvitePeople -> Unit // Todo Handle invite people action
             }
         }
 
+        LaunchedEffect(searchQuery) {
+            searchResults.value = if (MatrixPatterns.isUserId(searchQuery)) {
```

Comment:
Should we remove the first assignement expression and only keep the following code?
```
searchResults.value = if (searchQuery.isNotEmpty()) {
    performSearch(searchQuery)
} else {
    persistentListOf()
}
```

Reply:
Nope, the value is updated twice for this reason:

we want to update the search results as soon as the query has changed, if this is not a matrixId, we clear the list, otherwise, we display a "fake" result with this matrixId so the user would be able to start a chat with this matrixId.

Then, we perform the search query (this is an asynchronous operation that can take a few seconds) and we update the query results.

we'll also need some debounce for the search query but not on the first value update

Reply:
Okay I see, maybe we could add some comment to explain that, what do you think?